gdx/src/com/badlogic/gdx/graphics/g3d/shaders/BaseShader.java

```diff
@@ -113,6 +113,15 @@ public boolean validate (final BaseShader shader, final int inputID, final Rende
 	public Camera camera;
 	private Mesh currentMesh;
 
+	/** ClassX: thread-safety support */
+	public final Matrix4 tempMatrix4 = new Matrix4();
+
+	/** ClassX: thread-safety support */
+	public final Matrix3 tempMatrix3 = new Matrix3();
```

Comment:
`protected` also here?

Reply:
yep it should be. No real reason to make temp stuff public

Reply:
_tempMatrix3_ is the only one not referenced outside, so I can change the visibility to _protected_